### ISSUE TITLE
Embedded CSV Is Not Handled Correctly In PDF Conversion

### ISSUE THREAD

**Issue by jmwright** [Mach30/sliderule#2]
I was poking around with the [asciidoctor Docker script](https://github.com/Mach30/sliderule/blob/master/architecture/4-analysis/technical-demos/scripts/asciidoc-to-pdf.ps1). It looks like the embedding of files such as [CSV](https://github.com/Mach30/sliderule/blob/master/architecture/4-analysis/technical-demos/asciidoc-bom/bom.csv) will not work when converting to PDF. I can confirm that I just get a PDF with the raw text of the bom.adoc file in it. 

http://discuss.asciidoctor.org/Attachments-in-pdf-td6083.html

There's talk of using listing blocks, but there's a warning not to use it with "large" data sets. Whatever "large" means in that context. In any case, the use of embedded CSV data apparently isn't directly supported.

**Comment by m30-jrs** [comment]
Thanks for the heads up.  Testing that out was on my short list after getting DOF documented and putting together a couple of "post-it note" style example projects.  

Just a quick clarification...  Did you generate the PDF from the as committed bom.adoc file?  If so, it is not set up correctly yet.  I just dropped it in the repo so I would remember to come back and actually build the test correctly.  Below is literally the entire contents of bom.adoc

```
= To Do
J. Simmons
```

**Comment by jmwright** [comment]
> Did you generate the PDF from the as committed bom.adoc file?

No, I used a custom local setup just to verify that I could make it work, and then set the BoM up on the STEMN website. They have the concept of pipelines now, which enables pretty quick experimentation with things like this. I like where Sliderule is headed, and so I decided to play around a little bit with rendering BoMs and rolling the BoMs from component projects/subdirs up into the top level assembly BoM.

Here's the [initial Sliderule pipeline](https://stemn.com/files/5b15410c252c9400192ab940/5b19f2c1b2698100194c80cf?revision=5b1a0137b2698100194c827f) that just renders one component's BoM as PDF.

And here's the [super simple BoM](https://stemn.com/project/5b15410c252c9400192ab940/files/5b1750cdb1527c0019ec2202) that the pipeline is converting. It has the filament in the BoM per our discussion the other evening. The idea is to have the pipeline get the print volume from the CadQuery model, and use that to figure out approximately how many meters of filament are needed (making assumptions on things like layer height). When all the 3D printed parts are rolled up into the top level BoM, you know very roughly how much filament a project is going to take.

None of this is to bypass the post-it-note phase of testing. That whole project on STEMN is just me playing around with newer tech and some of the concepts that we've discussed. Pipelines make it pretty easy to test ideas out, and could be replaced with a wrapper script around git later for a Sliderule implementation. I could also create a Sliderule template project on STEMN with the directory structure and all the pipelines in place once Sliderule's methodology is more finalized. That could be just one of the many ways that people use Sliderule.

The idea of being able to do the equivalent of `npm install xxxx` for hardware is very exciting, so I've been spending some of my very few extra CPU cycles thinking and experimenting. I'm thinking through a way to store COTS parts in a separate project and do a sort of requirements.txt file with pipelines like Python does with pip. I'm not concerned about having to wipe the board and start over. In fact, I expect to have to as Sliderule progresses. I can't really grok and intelligently discuss new methodologies and tech though until I get hands-on with it. Inefficient iteration is just the price I have to pay.

**Comment by m30-jrs** [comment]
This is all very cool work.  By all means, keep up what you are doing.  Don't worry about getting ahead of me.  If I had the personal brain cycles I would have been trying to implement something like this myself to see what options we have for implementing BoMs in adoc files.  I also love the idea of automating these ideas in STEMN pipelines.  

Glad you are excited about the idea of `npm install myComponent`.  It is very encouraging.  It is certainly the reaction I am looking for.  Keep us posted on how your experiments go.  And look for more info about DOF this weekend.  I plan to do some more writing this afternoon.
